src/l3.c

```diff
@@ -20,13 +20,22 @@
 #include <sys/wait.h>
 #include <sys/syscall.h>
 #include <threads.h>
+#include <string.h>
+#include <assert.h>
 
 #include "l3.h"
 
 #ifdef L3_LOC_ENABLED
 #include "loc.h"
 #endif  // L3_LOC_ENABLED
 
+// Define version of static assertion checking that works for both gcc and g++
+#ifdef __cplusplus
+#define L3_STATIC_ASSERT    static_assert
+#else
+#define L3_STATIC_ASSERT    _Static_assert
```

Comment:
We need this redefinition as this file is compiled using `gcc` with .c samples, but compiled with `g++` when it's built along with C++ sample programs.